web_widget_html_markdown/static/src/js/web_widget_html_markdown.js

```diff
@@ -0,0 +1,861 @@
+/* global showdown */
+/* Copyright 2021 Therp B.V. - <http:///therp.nl>
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
+odoo.define("web_widget_html_markdown.FieldHtmlMarkDown", function(require) {
+    "use strict";
+
+    var ajax = require('web.ajax');
+    var basic_fields = require('web.basic_fields');
+    var config = require('web.config');
+    var core = require('web.core');
+    var Wysiwyg = require('web_editor.wysiwyg.root');
+    var field_registry = require('web.field_registry');
+    require('web._field_registry');
+    var Dialog = require('web.Dialog');
+    var QWeb = core.qweb;
+    var assetsLoaded;
+
+    var jinjaRegex = /(^|\n)\s*%\s(end|set\s)/;
+
+    var _t = core._t;
+    var _lt = core._lt;
+    var TranslatableFieldMixin = basic_fields.TranslatableFieldMixin;
+    var LIBS_PATH = "/web_widget_html_markdown/static/src/lib/";
+
+    /*
+        Events : _.extend({}, DebouncedField.prototype.events, {
+            'change .mk_html_switch' : '_switch_markdown_html',
+        }),
+    */
+    var FieldHtmlMarkDown = basic_fields.DebouncedField.extend(TranslatableFieldMixin, {
```

Comment:
So you've basically copied a lot of code from the original HTML widget of Odoo. And we're also using it, of course, when the user switches to Html.

To avoid duplication of code, we basically have two options:

1. Don't inherit from `DebouncedField`, but directly from HtmlField instead and then override its workings. I remember I specifically asked you to inherit from `DebouncedField` and it was because I saw some stuff in the Html widget's code that looked incompatible with what we wanted to do, but that was before we decided to offer live switching between HTML and Markdown. The situation might look different now - what's your thought on this?

2. Instead of creating a new widget, we just enhance the existing HTML widget so that ALL Html fields in the system can now be edited with Markdown. Seems on one hand very useful, but also a bit radical - at least user should be able to switch off that behaviour in order to have a remedy for any bugs that our enhancements might cause in his particular incarnation of Odoo.

I'm not bothered for now but let's keep this as an open discussion point for when we polish this module for OCA.